### PR TITLE
Add -p option to jslint command allowing results to be printed to stdout

### PR DESCRIPTION
**Pull request by mridgway**
Add a -p option to the `mojito jslint` command that prints the output to the console instead of to files. It does not change functionality for the default options.

`mojito jslint -p`

### CODE REVIEW

File: source/lib/management/commands/jslint.js [hunk 1]
```diff
@@ -122,6 +129,23 @@ function OutputFile(filename) {
...
 
 
+/*
+ * A very simple class to allow writes to stdout.
+ */
+function OutputStdout(filename) {
+    var printedName = false;
+
+    this.write = function(s) {
+        if (!printedName) {
```
Comment (drewfish) [inline source/lib/management/commands/jslint.js:139]: If I'm reading the code correctly, when -p is given the file _isn't_ written.  Is that correct?  If so, it seems strange to print the name of a file that doesn't exist.

**Comment by drewfish** [comment]
+1 to the idea

File: source/lib/management/commands/jslint.js (hunk 1 above)
Comment (mridgway) [inline source/lib/management/commands/jslint.js:139]: In this case, filename is the file that jslint was run on. The logic for instantiation differs slightly: https://github.com/yahoo/mojito/pull/199/files#L0R364

This variable ensures that I write the filename only if I write an error to the screen after instantiation.

**Comment by mridgway** [comment]
Example run on the getting-starting-guide example project:

```
mridgway:flickr-list mridgway$ mojito jslint app -p .
models/flickr.common.js
   1 7,32: Expected exactly one space between 'function' and '('.
     YUI.add('ModelFlickr', function(Y, NAME) {
   2 8,5: Missing 'use strict' statement.
     var API_KEY = '9cc79c8bf1942c683b0d4e30b838ee9c';
   3 12,34: Expected exactly one space between 'function' and '('.
     getFlickrImages: function(queryString, start, count, callback) {
   4 14,21: Missing radix parameter.
     start = parseInt(start) || 0;
   5 15,21: Missing radix parameter.
     count = parseInt(count) || 10;
   6 20,30: Expected exactly one space between 'function' and '('.
     Y.YQL(q, function(rawYqlData) {
   7 26,17: Expected 'rawPhoto' at column 21, not column 17.
     rawPhoto = null,
   8 27,17: Expected 'photos' at column 21, not column 17.
     photos = [],
   9 28,17: Expected 'photo' at column 21, not column 17.
     photo = null,
  10 29,17: Expected 'i' at column 21, not column 17.
     i = 0;
  11 31,24: Unexpected space between ';' and 'i'.
     for (; i < rawPhotos.length; i++) {
  12 31,24: Expected ';' and instead saw 'i'.
     for (; i < rawPhotos.length; i++) {
  13 31,26: Unexpected space between 'i' and '<'.
     for (; i < rawPhotos.length; i++) {
  14 31,26: Expected ')' and instead saw '<'.
     for (; i < rawPhotos.length; i++) {
  15 31,28: Expected '{' and instead saw 'rawPhotos'.
     for (; i < rawPhotos.length; i++) {
  16 31,38: Expected an assignment or function call and instead saw an expression.
     for (; i < rawPhotos.length; i++) {
  17 31,47: Unexpected '++'.
     for (; i < rawPhotos.length; i++) {
  18 31,49: Expected ';' and instead saw ')'.
     for (; i < rawPhotos.length; i++) {
  19 31,49: Expected an identifier and instead saw ')'.
     for (; i < rawPhotos.length; i++) {
  20 31,49: Stopping.  (41% scanned).

mojits/FlickrBrowser/assets/logmutator.js
   1 7,42: Expected exactly one space between 'function' and '('.
     YUI.add('flickr-logger-mutator', function(Y) {
   2 30,1: Missing 'use strict' statement.
     }, 'whatever');
mojits/FlickrBrowser/binders/index.js
   1 7,45: Expected exactly one space between 'function' and '('.
     YUI.add('FlickrBrowserBinderIndex', function(Y, NAME) {
   2 25,5: Missing 'use strict' statement.
     Y.namespace('mojito.binders')[NAME] = {
   3 31,23: Expected exactly one space between 'function' and '('.
     init: function(mojitProxy) {
   4 33,67: Expected exactly one space between 'function' and '('.
     this.mojitProxy.listen('flickr-image-chosen', function(event) {
   5 47,23: Expected exactly one space between 'function' and '('.
     bind: function(node) {}
mojits/FlickrBrowser/controller.common.js
   1 7,34: Expected exactly one space between 'function' and '('.
     YUI.add('FlickrBrowser', function(Y) {
   2 21,5: Missing 'use strict' statement.
     Y.mojito.controller = {
   3 29,24: Expected exactly one space between 'function' and '('.
     index: function(ac) {

...

85 errors found.
```

File: source/lib/management/commands/jslint.js (hunk 1 above)
Comment (drewfish) [inline source/lib/management/commands/jslint.js:139]: Ah great, I misunderstood.

**Comment by FabianFrank** [comment]
+1 great idea

**Pull request merged**